journal/src/main/java/io/camunda/zeebe/journal/file/SparseJournalIndex.java

```diff
@@ -18,12 +18,15 @@
 import io.camunda.zeebe.journal.JournalRecord;
 import java.util.Map;
 import java.util.TreeMap;
+import java.util.concurrent.ConcurrentNavigableMap;
+import java.util.concurrent.ConcurrentSkipListMap;
 
 final class SparseJournalIndex implements JournalIndex {
 
   private final int density;
-  private final TreeMap<Long, Integer> indexToPosition = new TreeMap<>();
-  private final TreeMap<Long, Long> asqnToIndex = new TreeMap<>();
+  private final ConcurrentNavigableMap<Long, Integer> indexToPosition =
```

Comment:
:wrench: As we only update some things from multiple threads, maybe we want to annotate/document that (i.e. why is `indexToAsqn` not thread-safe). I don't know how likely this is to cause confusion in the future...though it took me a good 10 seconds to understand why this change, but maybe that's just me :smile:

Reply:
Actually, I think `indexToAsqn` should also be thread safe. Now, the readers while seeking can update the it. :face_with_head_bandage: